### PR TITLE
Release 0.58.0

### PR DESCRIPTION
**Pull request by paulfantom**
Signed-off-by: Paweł Krupa (paulfantom) <pawel@krupa.net.pl>## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+## 0.58.0 / 2022-07-19
+
+* [FEATURE] Add validations for timeout and time settings of alertmanager at CRD level. #4827, #4881
+* [FEATURE] Extend the PrometheusSpec to allow to configure the `max_exemplars`. #4834
+* [FEATURE] Add support for web TLS configuration for Alertmanager CRD. #4868
+* [ENHANCEMENT] Optimise warning log message during sanitization of OpsGenie configuration. #4833
```
Comment (simonpasquier) [inline CHANGELOG.md:6]: [BUGFIX] rather?

File: go.mod [hunk 2]
```diff
@@ -107,33 +104,27 @@ require (
...
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 replace (
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring => ./pkg/apis/monitoring
 	github.com/prometheus-operator/prometheus-operator/pkg/client => ./pkg/client
-	// A replace directive is needed for github.com/prometheus/alertmanager
-	// because Cortex (which is an indirect dependency through Thanos) has a
-	// requirement on a more recent commit version but untagged.
-	github.com/prometheus/alertmanager => github.com/prometheus/alertmanager v0.24.0
```
Comment (simonpasquier) [inline go.mod:133]: are you sure that we can remove the replace for Alertmanager?

Comment (paulfantom) [reply go.mod:133]: Considering that override was from 0.24.0 to 0.24.0, I would say that we can do this.

Check https://github.com/prometheus-operator/prometheus-operator/pull/4907/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L26

**Review by simonpasquier — APPROVED** [review]
:shipit:

**Pull request merged**